Zend/tests/pattern_matching/is/object.phpt

```diff
@@ -0,0 +1,127 @@
+--TEST--
+Object pattern matching
+--FILE--
+<?php
+
+class Foo {
+    public function __construct(
+        public int $a,
+    ) {}
+
+    public static function isSelfFoo($value) {
+        return $value is self {};
+    }
+
+    public static function isStaticFoo($value) {
+        return $value is static {};
+    }
+}
+
+class Bar extends Foo implements Baz {
+    public static function isSelfBar($value) {
+        return $value is self {};
+    }
+
+    public static function isParentBar($value) {
+        return $value is parent {};
+    }
+}
+
+interface Baz {}
+class Qux {}
+
+$foo = new Foo(42);
+$bar = new Bar(43);
+$qux = new Qux();
+
+var_dump($foo is Foo {});
```

Comment:
I'm unclear why the {} is needed after a class name.  As long as we ignore global constants, isn't it sufficiently non-ambiguous?

Reply:
It's not, we're testing the object pattern here, removing the `{}` would not make it an object pattern.

Reply:
Hm.  So what's the practical difference between `$foo is Foo` and `$foo is Foo{}` ?  Without any property restrictions or binding it looks like it would be effectively identical, yes?

Reply:
Yes, but there's no reason to arbitrarily require properties to be matched. We don't do that anywhere else in the language. E.g. you can create empty ifs, switches, etc.

Reply:
OK, so in practice most people would use `$foo is Foo`, but this test is just confirming that the null-case of object matching doesn't do anything weird.

PHPT files need more docs. 😄